src/main/webapp/components/navbar.vue

```diff
@@ -0,0 +1,167 @@
+<template>
+  <nav class="blue sticky nav-bar">
+    <a href="index.html" class="bold" id="title">starfish</a>
+    <img src="../assets/starfish.png" id="logo">
+    <div id="action-items">
+      <a href="#" id="post-action">PostANote</a>
+      <div class="dropdown">
+        <button class="dropbtn" @click.stop="toggleDropdown">
+          <img id="user-profile" src="../assets/user.svg">
+        </button>
+        <div class="light-gray dropdown-content" v-if="showDropdown" v-click-outside="hideDropdown">
+          <img src="../assets/user.svg" height="115px"/>
+          <p class="bold" id="dropdown-name">{{ user.name }}</p>
+          <p id="dropdown-points">{{ user.points }} points</p>
+          <a class="dropdown-link" href="#">My Profile</a> 
+          <a class="dropdown-link" href="#">Favorite Notes</a>
+          <a class="dropdown-link" href="#">Logout</a>
+        </div>
+      </div>
+    </div> 
+  </nav>
+</template>
+
+<script>
+Vue.directive('click-outside', {
+  bind: function (el, binding, vnode) {
+    this.event = function(event) {
+      if(!(el == event.target || el.contains(event.target))) {
+        vnode.context[binding.expression](event);
+      }
+    };
+    document.body.addEventListener('click', this.event);
+  },
+  unbind: function(el) {
+    document.body.removeEventListener('click', this.event);
+  }
+});
+
+module.exports = {
+  data: function() {
+    return {
+      showDropdown: false
+    }
+  },
+  props: {
+    user: Object
+  },
+  methods: {
+    hideDropdown() {
+      this.showDropdown = false;
+    },
+    toggleDropdown() {
+      this.showDropdown = !this.showDropdown;
+    }
+  }
+}
+</script>
+
+<style scoped>
+.bold {
+  font-weight: bold;
+}
+
+.blue {
+  background-color: #004aad;
+}
+
+.light-gray {
+  background-color: #f9f9f9;
+}
+
+.sticky {
+  position: fixed;
+  top: 0;
+  width: 100%;
+}
+
+.nav-bar {
+  align-items: center;
+  display: flex;
+  font-family: 'Roboto';
+  height: 80px;
+}
+
+.nav-bar a {
```

Comment:
Consider using scss notation to nest css properties:

https://flaviocopes.com/vue-css/

Could be applied to combine this rule with the above rule

Reply:
Also could apply to some other rules in this file.

Reply:
Ok got it, I'll look into adding scss into the project!

Reply:
I _think_ vuetify already incorporates scss or sass, but based on [their page on SASS](https://vuetifyjs.com/en/customization/sass-variables/) it's not clear to me how we would use it in our components.

Reply:
Do you mean why we would use scss in our components?

Reply:
No, sorry, I meant that since we're not using npm like the page mentions, I can't tell if we need to set up something else before we can start using scss 😆

Reply:
Ohh ok I see, it does seem like we need to do something else to get it set up